sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1068,6 +1068,14 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val ARROW_FALLBACK_ENABLE =
+    buildConf("spark.sql.execution.arrow.fallback.enabled")
+      .doc("When true, the optimization by 'spark.sql.execution.arrow.enabled' " +
+        "could be disabled when it is unable to be used, and fallback to " +
+        "non-optimization.")
```

Comment:
Just a suggestion: "When true, optimizations enabled by 'spark.sql.execution.arrow.enabled' will fallback automatically to non-optimized implementations if an error occurs."